pkgs/development/libraries/dnnl/default.nix

```diff
@@ -22,12 +22,18 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
+  doCheck = true;
+
   # The test driver doesn't add an RPath to the build libdir
   preCheck = ''
     export LD_LIBRARY_PATH=$PWD/src
   '';
 
-  doCheck = true;
+  # The cmake install gets tripped up and installs a nix tree into $out, in
+  # addition to the correct install; clean it up.
+  postInstall = ''
+    rm -r $out/nix
+  '';
```

Comment:
On master, we seem to have gained a superfluous install tree:
```
$ nix build -f . dnnl.{out,dev,doc}; tree result*
result
├── lib
│   ├── libdnnl.so -> libdnnl.so.1
│   ├── libdnnl.so.1 -> libdnnl.so.1.2
│   └── libdnnl.so.1.2
└── nix
    └── store
        └── wmf3lp0bxz8r0vnyfx2zrg9ynhhrn9al-dnnl-1.2
            └── lib
                ├── cmake
                │   └── mkldnn
                ├── libmkldnn.so -> libdnnl.so
                ├── libmkldnn.so.1 -> libdnnl.so.1
                └── libmkldnn.so.1.2 -> libdnnl.so.1.2
result-1-dev
├── include
│   ├── dnnl_config.h
│   ├── dnnl_debug.h
│   ├── dnnl.h
│   ├── dnnl.hpp
│   ├── dnnl_types.h
│   ├── dnnl_version.h
│   ├── mkldnn_config.h
│   ├── mkldnn_debug.h
│   ├── mkldnn_dnnl_mangling.h
│   ├── mkldnn.h
│   ├── mkldnn.hpp
│   ├── mkldnn_types.h
│   └── mkldnn_version.h
├── lib
│   └── cmake
│       └── dnnl
│           ├── dnnl-config.cmake
│           ├── dnnl-config-version.cmake
│           ├── dnnl-targets.cmake
│           └── dnnl-targets-release.cmake
└── nix-support
    └── propagated-build-inputs
result-2-doc
└── share
    └── doc
        └── dnnl
            ├── LICENSE
            └── README

15 directories, 26 files
```

On this PR:
```
$ nix build -f . dnnl.{out,dev,doc}; tree result*
result
└── lib
    ├── libdnnl.so -> libdnnl.so.1
    ├── libdnnl.so.1 -> libdnnl.so.1.2
    └── libdnnl.so.1.2
result-1-dev
├── include
│   ├── dnnl_config.h
│   ├── dnnl_debug.h
│   ├── dnnl.h
│   ├── dnnl.hpp
│   ├── dnnl_types.h
│   ├── dnnl_version.h
│   ├── mkldnn_config.h
│   ├── mkldnn_debug.h
│   ├── mkldnn_dnnl_mangling.h
│   ├── mkldnn.h
│   ├── mkldnn.hpp
│   ├── mkldnn_types.h
│   └── mkldnn_version.h
├── lib
│   └── cmake
│       └── dnnl
│           ├── dnnl-config.cmake
│           ├── dnnl-config-version.cmake
│           ├── dnnl-targets.cmake
│           └── dnnl-targets-release.cmake
└── nix-support
    └── propagated-build-inputs
result-2-doc
└── share
    └── doc
        └── dnnl
            ├── LICENSE
            └── README

9 directories, 23 files
```

Reply:
Likely an upstream bug. Perhaps some files are installed assuming a relative path.

A similar issue we had with also a cmake build in #76855.